pfhedge/nn/modules/bs/european.py

```diff
@@ -89,15 +101,50 @@ def extra_repr(self) -> str:
         params.append("strike=" + _format_float(self.strike))
         return ", ".join(params)
 
+    def _acquire_params_from_derivative(
+        self,
+        log_moneyness: Optional[Tensor] = None,
+        time_to_maturity: Optional[Tensor] = None,
+        volatility: Optional[Tensor] = None,
+    ) -> Tuple[Tensor, Tensor, Tensor]:
+        if log_moneyness is None:
+            if self.derivative is None:
+                raise ValueError(
+                    "log_moneyness is required if derivative is not set at this initialization."
+                )
+            if self.derivative.ul().spot is None:
+                raise AssertionError("please simulate first")
```

Comment:
Which error you recommend?

Reply:
Sorry I missed this. Not exactly sure but I would choose AttributeError since an attribute reference to .spot is failing.
https://docs.python.org/3/library/exceptions.html#AttributeError